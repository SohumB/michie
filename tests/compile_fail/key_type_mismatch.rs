use caching::caching;

#[caching(key_type = (), key_expr = a)]
fn f(a: bool) -> bool {
    a
}

fn main() {}
