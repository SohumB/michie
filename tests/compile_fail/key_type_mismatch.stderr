error[E0308]: mismatched types
 --> tests/compile_fail/key_type_mismatch.rs:3:38
  |
3 | #[memoized(key_type = (), key_expr = a)]
  |                                      ^ expected `()`, found `bool`
  |
  = note: expected reference `&()`
             found reference `&bool`
  = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> tests/compile_fail/key_type_mismatch.rs:3:38
  |
3 | #[memoized(key_type = (), key_expr = a)]
  |                                      ^ expected `()`, found `bool`
  |
  = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)
